src/path.ts

```diff
@@ -682,6 +682,14 @@ export class PathRef {
     }
     return this;
   }
+
+  startsWith(token: string): boolean {
+    return this.#path.startsWith(token);
```

Comment:
It's not so simple. It should act similar to Rust's startsWith and endsWith so the following would occur:

```
const dir = $.path("/some/directory");
assert(!dir.startsWith("/some/dir");
```

More details:

https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.starts_with
https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.ends_with

Probably we should implement `.components()` (https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.components) first as that will make it easier.

Reply:
Thanks for clarifying, it indeed seemed to simple to be true !

I'll take a look at everything !